This patch incorporates two commits in the tree at the Gnome Repository
that were never released.  It fixes the problem of gtksourceview2 failing to build under 
Mavericks and later when both gtk2 and gtk3 macports are installed.
This version of the patch uses the upstream package naming conventions that
are canon as of August 2015.

From e88357c5f210a8796104505c090fb6a04c213902 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Mon, 21 Nov 2011 22:26:21 +0100
Subject: Change IgeMacIntegration to GtkOSXApplication

Instead of using the deprecated IgeMacIntegration which doesn't work
on 32-bit Macs, get the locale dir using the quartz_application_...
API from GtkOSXApplication.

From e38712a2061f6f45116f6316fd32cc166c68baf6 Mon Sep 17 00:00:00 2001
From: Pierre Boutillier <pierre.boutillier@ens-lyon.org>
Date: Tue, 6 Nov 2012 15:49:43 +0100
Subject: Update to gtk-mac-integration 2.0 API

Fixes bgo#687766. quartz_application_... becomes
gtkosx_application_...

https://bugzilla.gnome.org/show_bug.cgi?id=687766


diff --git configure.ac configure.ac
index ccfe986..42fe313 100644
--- configure.ac
+++ configure.ac
@@ -76,7 +76,7 @@ AM_CONDITIONAL(OS_OSX, test "$os_osx" = "yes")
 if test "$os_osx" = "yes"; then
 	AC_DEFINE([OS_OSX],[1],[Defined if os is Mac OSX])
 
-	PKG_CHECK_MODULES(IGE_MAC, ige-mac-integration)
+	PKG_CHECK_MODULES(GTK_MAC, gtk-mac-integration-gtk2 >= 2.0.0)
 fi
 
 # Check for Glade3
diff --git gtksourceview/Makefile.am gtksourceview/Makefile.am
index f3781d5..d1136a4 100644
--- gtksourceview/Makefile.am
+++ gtksourceview/Makefile.am
@@ -10,7 +10,7 @@ INCLUDES = 				\
 	-I$(top_srcdir) -I$(srcdir) 	\
 	$(DISABLE_DEPRECATED)		\
 	$(WARN_CFLAGS) 			\
-	$(IGE_MAC_CFLAGS)		\
+	$(GTK_MAC_CFLAGS)		\
 	$(DEP_CFLAGS)
 
 BUILT_SOURCES = 			\
@@ -96,7 +96,7 @@ nodist_libgtksourceview_2_0_la_SOURCES =\
 completion_providers = 							\
 	completion-providers/words/libgtksourcecompletionwords.la
 
-libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS) $(IGE_MAC_LIBS) $(completion_providers)
+libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS) $(GTK_MAC_LIBS) $(completion_providers)
 libgtksourceview_2_0_la_LDFLAGS = -no-undefined -export-symbols-regex "^gtk_source_.*"
 libgtksourceview_2_0_includedir = $(includedir)/gtksourceview-2.0/gtksourceview
 
diff --git gtksourceview/gtksourceview-i18n.c gtksourceview/gtksourceview-i18n.c
index e4db3eb..32f8068 100644
--- gtksourceview/gtksourceview-i18n.c
+++ gtksourceview/gtksourceview-i18n.c
@@ -24,7 +24,7 @@
 #endif
 
 #ifdef OS_OSX
-#include <ige-mac-bundle.h>
+#include <gtkosxapplication.h>
 #endif
 
 #include <string.h>
@@ -45,11 +45,9 @@ get_locale_dir (void)
 
 	g_free (win32_dir);
 #elif defined (OS_OSX)
-	IgeMacBundle *bundle = ige_mac_bundle_get_default ();
-
-	if (ige_mac_bundle_get_is_app_bundle (bundle))
+	if (gtkosx_application_get_bundle_id () != NULL)
 	{
-		locale_dir = g_strdup (ige_mac_bundle_get_localedir (bundle));
+		locale_dir = g_build_filename (gtkosx_application_get_resource_path (), "share", "locale", NULL);
 	}
 	else
 	{
diff --git a/tests/Makefile.am b/tests/Makefile.am
index dfca414..1c6085b 100644
--- tests/Makefile.am
+++ tests/Makefile.am
@@ -78,6 +78,7 @@ test_regex_SOURCES = test-regex.c
 test_regex_LDADD = 						\
 	$(top_builddir)/gtksourceview/libgtksourceview-private.la	\
 	$(DEP_LIBS)			\
+	$(GTK_MAC_LIBS)		\
 	$(TESTS_LIBS)
 
 UNIT_TEST_PROGS += test-completion-model
-- 
cgit v0.10.1

